src/algebra/group/with_one.lean

```diff
@@ -74,14 +74,21 @@ section lift
 @[to_additive]
 def lift {β : Type v} [has_one β] (f : α → β) : (with_one α) → β := λ x, option.cases_on x 1 f
 
-variables [semigroup α] {β : Type v} [monoid β] (f : α → β)
+variables [ha : semigroup α] {β : Type v} [monoid β] (f : α → β)
```

Comment:
How about just not including `[semigroup α]` in this line at all, and then put it where you have `include ha`.

Reply:
The goal of this `[ha :` was to preserve the order of implicit arguments in other definitions. I have no preferences between your and my versions.

Reply:
Oh I see, that seems okay then.